Add support for consent disagreement

slavklenov opened the pull request:
Added AbortSession function to MPinSDK.

Closes: MAASMOB-435

slavklenov committed:
Add support for consent disagreement

Added AbortSession function to MPinSDK.

Closes: MAASMOB-435

slavklenov committed:
Merge remote-tracking branch 'github-apache/master'

slavklenov committed:
Refactor the SDK support for the MFA platform

Added two SDK classes - MPinSDK for the old SDK and MfaSDK for the new MFA flow.
Moved common functionality to a common base class MPinSDKBase.
Added customerId and appId to a User.
Changed users storage to be organized by mpinId only (and not by backend).

Closes: MAASMOB-436, MAASMOB-442

slavklenov committed:
Refactor the SDK support for the MFA platform

Added two SDK classes - MPinSDK for the old SDK and MfaSDK for the new MFA flow.
Moved common functionality to a common base class MPinSDKBase.
Added customerId and appId to a User.
Changed users storage to be organized by mpinId only (and not by backend).

Closes: MAASMOB-436, MAASMOB-442

Commit:
Resolve conflict with LLVM compiler

slavklenov committed:
Changed "customerIconURL" parameter name to "customerLogoURL".
Moved MfaSDK contstructor body from mfa_sdk.h to mfa_sdk.cpp

slavklenov committed:
Added test for MfaSDK.
Removed MFA related code from MPinSDK test.

slavklenov committed:
Add support for logging into the Mobile App

Added GetAccessCode method to MfaSDK.
Added accessCode and authzCode parameters to FinishAuthentication.
Renamed FinishAuthenticationAC to FinishAuthentication.
Return new SDK status code BAD_CLIENT_VERSION when GetServiceDetails receives 412 http status.

Closes: MAASMOB-475, MAASMOB-476

slavklenov committed:
Add support for trusted domains

Added URL parsing functionality.
Added url checking against a predefined list of trusted domains.
Added functions for trusted domains list manipulation.
Added tests for the URL parsing finctionality.
Updated MfaSDK cmd line test to do the URL parsing tests, to test the trusted domains functionality and to automatically request access codes.

Closes: MAASMOB-467

slavklenov committed:
Add users JSON versioning.

slavklenov committed:
Added URL parsing tests with unicode (non-latin language) urls. Added functionality for proper string to lower case convertion for arbitrary utf-8 strings (disabled by default).

slavklenov committed:
Added authzRequest parameter to authenticate request.

slavklenov committed:
Added IsUserExistimg function to MPinSDKBase.
Changed unique user key to contain customerId and appId.
Fixed bug, allowing to register user with the same key.
Added crypto storage clearing of old ids.
Updated mfa_sdk_test with restart registration test.

slavklenov committed:
Fixed MPinSDKBase::RestartRegistration for the MFA flow by saving and resending accessCode.

slavklenov committed:
Changed user id in mfa_sdk_test.

slavklenov committed:
Updated all tests code and naming to be consistent with the new SDK structure.
Removed old visual studion 2012 projects.

slavklenov committed:
Added unit tests for MfaSDK.
Fixed several potential problems in MfaSDK and MPinSDKBase.

slavklenov committed:
Added GetService/SessionDetails, URL parsing and trusted domains tests.

Commit:
Add granular docs

Commit:
Add MfaSDK documentation

slavklenov committed:
Fixed MfaSDK unit tests compilation issue.

slavklenov committed:
Fixed MfaSDK unit tests compilation issue.

slavklenov committed:
Replaced cv_shared_ptr with c++11 std::shared_ptr.

slavklenov committed:
Enabled c++11 in old MPinSDK unit tests Makefile.

Commit:
Travis Integration (#1)

* Initial .travis.yml
* Fix format error
* Fix build
* Add coverage

Commit:
Modify crypto params (#2)

Remove GLV

slavklenov committed:
Added OTP authentication to MfaSDK.

slavklenov committed:
Fixed MfaSDK OTP authentication by adding StartAuthenticationOTP method.

slavklenov committed:
Added unit tests for MfaSDK OTP authentication.
Fixed base authentication flow to pass only authOTT param when authenticating against RPA.
Fixed MPinSDK OTP authentication unit tests and test recorded data.

slavklenov committed:
Add REGISTRATION_EXPIRED status code

Return REGISTRATION_EXPIRED mpin status when making rps/user PUT or rps/signature GET requests and the backend returns 404 HTTP status code.

Closes: MAASMOB-548

simeon-aladjem reviewed (changes requested):
As for now, we are not merging this code to Apache. Please cancel the PR!

slavklenov committed:
Add regOTT expiration information (#3)

Add regOTT expiration information

slavklenov committed:
Migrate to milagro-crypto-c

Closes: MAASMOB-547

slavklenov committed:
Merge pull request #4 from miracl/milagro-crypto-c-integration

Migrate to milagro-crypto-c

slavklenov committed:
Coverage (#5)

* Added local coverage generation script.
Removed src/utf8 from coverage stats.

* Increased coverage.
Updated coverage generation script.
Fixed bug with getting time permits from S3.

slavklenov committed:
Updated to milagro-crypto-c master

slavklenov committed:
Migrated windows projects to visual studio 2017.
Replaced tabs with spaces in src files.
